src/app/server/Server.cpp

```diff
@@ -162,10 +170,18 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     SetAttributePersistenceProvider(&mAttributePersister);
     SetSafeAttributePersistenceProvider(&mAttributePersister);
 
-    // Ember requires a persistence provider:
-    //   - InitDataModelHandler uses ember init (so needs a data model provider)
-    //   - CodegenDataModelProvider requires a persistence provider to use, so this must be called late
-    //     enough, after SetSafeAttributePersistenceProvider
+    // Until all ember calls are handled by the data model provider ensure that this is done
+    // early enough.
+    // In particular ember initialization (emberAfInit) is done in InitDataModelHandler which happens
+    // below and may load NVM attributes, so data model should be initialized (e.g. as we transition to
+    // having data model provider initialization starting up ember prerequisites). The call stack is:
+    //    - InitDataModelHandler -> emberAfInit -> emberAfInitializeAttributes -> emAfLoadAttributeDefaults
+    //      may load from persistent storage.
+    //
+    // As we expect the CodegenDatamodelProvider to everntually be responsible for ember attribute
+    // persistence initialization, we place this:
+    //   - After SetSefeAttributePersistenceProvider (as this is the global persistence provider)
+    //   - Before InitDataModelHandler to allow data model startup to set up persistence for ember as needed
```

Comment:
OK, but the plan is presumably to have InitDataModelHandler go away altogether, because literally the only thing it does is start up the ember bits, right?

Reply:
Yes, all ember functions should only ever be used or seen by the codegen model. Ember is a possible implementation of the data model provider.